src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerIntellisenseFilter.cs

```diff
@@ -38,12 +40,37 @@ internal class DebuggerIntelliSenseFilter<TPackage, TLanguageService> : Abstract
 
         internal void SetNextFilter(IOleCommandTarget nextFilter)
         {
-            this.NextCommandTarget = nextFilter;
+            _originalNextCommandFilter = nextFilter;
+            SetNextFilterWorker();
+        }
+
+        private void SetNextFilterWorker()
+        {
+            // We have a new _originalNextCommandFilter or new _context, reset NextCommandTarget chain based on their values.
+            // The chain is formed like this: 
+            // this.CurrentHandlers (legacy command handlers) 
+            //     -> IVsCommandHandlerServiceAdapter (our command handlers migrated to the modern editor commanding)
+            //            -> original next command filter
+            var nextCommandFilter = _originalNextCommandFilter;
+            if (_context != null)
```

Comment:
I admit I don't quite get this check here....can we add a comment to explain it? Is this some thing where we create one of these prior to having the buffer, and that's what the later hookup is for?